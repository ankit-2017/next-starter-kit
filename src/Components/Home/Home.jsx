import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import Styles from './Home.module.scss';

function Home({ ip }) {
  return (
    <div className={Styles.homeContainer}>
      <h1>
        Your Public Ip:
        {ip}
      </h1>
      <Link href="/other"> go to other </Link>
    </div>
  );
}

Home.defaultProps = {
  ip: '',
};

Home.propTypes = {
  ip: PropTypes.string,
};

export default Home;
