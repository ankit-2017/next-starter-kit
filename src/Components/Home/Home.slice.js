import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axiosClient from 'Utils/axiosClient';

export const getPublicIP = createAsyncThunk(
    'getPublicIp',
    async (data, { rejectWithValue }) => {
        try {
            const res = await axiosClient({ url: 'https://api.ipify.org/?format=json', method: 'get' });
            return res.data;
        } catch (error) {
            return rejectWithValue(error?.response?.data);
        }
    },
);

const getPublicIpSlice = createSlice({
    name: 'public-ip',
    initialState: {
        isFetching: false,
        error: null,
        data: {},
    },
    extraReducers: {
        [getPublicIP.pending]: (state) => {
            state.isFetching = true;
        },
        [getPublicIP.fulfilled]: (state, { payload }) => {
            state.isFetching = false;
            state.data = payload;
        },
        [getPublicIP.rejected]: (state, { payload, error }) => {
            state.isFetching = false;
            state.error = payload?.message || error?.message;
        },
    },
});

export default getPublicIpSlice.reducer;
