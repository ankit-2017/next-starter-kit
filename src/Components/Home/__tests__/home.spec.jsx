import { render } from '@testing-library/react'
import Home from '../Home'

describe('<Home/>', ()=> {
    it('Should render home component', () => {
        const props = {
            ip: 'test IP'
        }
        const { asFragment } = render(<Home {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })
})