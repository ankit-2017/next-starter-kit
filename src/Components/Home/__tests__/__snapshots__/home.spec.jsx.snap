// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Should render home component 1`] = `
<DocumentFragment>
  <div
    class="homeContainer"
  >
    <h1>
      Your Public Ip: test IP
    </h1>
  </div>
</DocumentFragment>
`;
